server/images.go

```diff
@@ -228,27 +228,32 @@ func GetModel(name string) (*Model, error) {
 	return model, nil
 }
 
-func realpath(p string) string {
-	abspath, err := filepath.Abs(p)
+func realpath(mfPath, from string) string {
+	abspath, err := filepath.Abs(from)
 	if err != nil {
-		return p
+		return from
 	}
 
 	home, err := os.UserHomeDir()
 	if err != nil {
 		return abspath
 	}
 
-	if p == "~" {
+	if from == "~" {
 		return home
-	} else if strings.HasPrefix(p, "~/") {
-		return filepath.Join(home, p[2:])
+	} else if strings.HasPrefix(from, "~/") {
+		return filepath.Join(home, from[2:])
+	}
+
+	if _, err := os.Stat(filepath.Join(mfPath, from)); err == nil {
+		// this is a file relative to the Modelfile
+		return filepath.Join(mfPath, from)
 	}
 
 	return abspath
 }
 
-func CreateModel(ctx context.Context, name string, commands []parser.Command, fn func(resp api.ProgressResponse)) error {
+func CreateModel(ctx context.Context, name, modelFilePath string, commands []parser.Command, fn func(resp api.ProgressResponse)) error {
```

Comment:
this var name is no longer appropriate since it's the directory containing the modelfile rather than the modelfile itself

```suggestion
func CreateModel(ctx context.Context, name, modelfileDir string, commands []parser.Command, fn func(resp api.ProgressResponse)) error {
```